### ISSUE TITLE
deactivating/canceling freehand while drawing is not working

### ISSUE THREAD

**Issue by Zaid-Safadi** [cornerstonejs/cornerstoneTools#784]
### Prerequisites

* [ ] Are you running the latest version?
V 2.4.0
* [x] Are you reporting to the correct repository?
* [x] Did you perform a cursory search?

### Description
Trying to implement canceling a freehand while drawing is causing an error in the method: 
[closeToolIfDrawing](https://github.com/cornerstonejs/cornerstoneTools/blob/master/src/imageTools/freehand.js#L1126-L1134)
```
function closeToolIfDrawing(element) {
  const config = freehand.getConfiguration();
  if (config.currentTool >= 0) {
    // Actively drawing but changed mode.
    const lastHandlePlaced = config.currentHandle;

    endDrawing(element, lastHandlePlaced);
  }
}
```
The method is passing the wrong parameters to the method "endDrawing"

Also, while the "endDrawing" tries to close the handles of not completed. It doesn't take into account if there is only "1" handle 

### Steps to Reproduce

1. Capture the keyboard "Esc" key event 
2. Write code to Deactivate and re-activate the tool 
3. while drawing a freehand, press the Esc key and see the error in the debugger.

**Expected behavior:** [What you expected to happen]
Freehand will be either deleted if only handle is drawn or closed if possible.

**Actual behavior:** [What actually happened]
Error in the debugger

### Codepen With Reproduction of Issue:
```
    element.addEventListener("cornerstonetoolskeydown", onKeyDown);
    function onKeyDown(e) {
      const data = e.detail;
      const keyCode = data.keyCode;

      if (keyCode === keyCodes.ESC) {
        cornerstoneTools.freehand.deactivate(element, 1);
        cornerstoneTools.freehand.activate(element, 1);
      }
```

**Comment by Zaid-Safadi** [comment]
This is fixed for v2 in branch v2.4.x.
I will leave it open to apply the fix for v3

**Comment by dannyrb** [comment]
Fixed here: https://github.com/cornerstonejs/cornerstoneTools/pull/880
